tests/cupy_tests/random_tests/common_distributions.py

```diff
@@ -268,3 +268,23 @@ def test_gamma_2(self):
     def test_gamma_ks(self):
         self.check_ks(0.05)(
             self.shape, self.scale, size=2000)
+
+
+geometric_params = [
+    {'p': 0.5},
+    {'p': 0.1},
+    {'p': 1.0},
```

Comment:
test also arrays please

Reply:
I tried passing the array as test parameters, but as of now we have to explicitly pass p as cupy.ndarray for distributions of Generator, but in RandomState there is no such requirement. And tests fails due to broadcast error.

Reply:
> And tests fails due to broadcast error.

Maybe the shape of `p` and `size` make a conflict?

Reply:
https://github.com/cupy/cupy/blob/d53212bb87579013ec8a1c4b8d31de8512f92b40/tests/cupy_tests/random_tests/common_distributions.py#L111-L116

Is there any specific se of defining the numpy_len = 1000? 
As this is causing the broadcast error. I replaced the above line with following and it works well for the p as ndarray, but was failing for the `lognormal` and `normal` by rejecting null hypothesis.

```python
# kwargs['size'] = numpy_len
dtype = kwargs.pop('dtype', None)
numpy_rng = self.get_rng(numpy, self.__seed)
func_numpy = getattr(numpy_rng, self.target_method)(*args, **kwargs)
vals_numpy = func_numpy
count = 1 + (numpy_len - 1) // vals_numpy.size
for _ in range(1, count):
    vals_numpy = numpy.append(vals_numpy, func_numpy)

if dtype is not None:
    vals_numpy = vals_numpy.astype(dtype, copy=False)
vals_numpy = numpy.stack(vals_numpy).ravel()
```

Note: This change was only for testing the distribution of RandomState as for Generator `p` is required to be cupy.ndarray or a scalar.

Reply:
I don't think `check_ks` for `p` of ndarray makes much sense, so we can skip it.